02_cancer_type_classification/lasso_range_analysis/nbconverted/lasso_range_all.py

```diff
@@ -0,0 +1,280 @@
+sns.set({'figure.figsize': (28, 6)})
+sns.set_style('ticks')
+
+def print_corr_name(correlation):
+    return correlation.upper() if correlation == 'ccc' else correlation.capitalize()
+
+# order boxes by median pearson per gene
+gene_order = (corr_cancer_type_df
+    .groupby('gene')
+    .agg(np.median)
+    .sort_values(by=corr_column, ascending=False)
+).index.values
+
+with sns.plotting_context('notebook', font_scale=1.5):
+    ax = sns.boxplot(data=corr_cancer_type_df, order=gene_order, x='gene', y=corr_column)
+    ax.axhline(0.0, linestyle='--', color='grey')
+    plt.xticks(rotation=90)
+    plt.title(f'Model size/performance correlations across cancer types, per gene (nonzero cutoff: {nz_coefs_cutoff:.0f})', y=1.02)
+    plt.xlabel('Gene')
+    plt.ylabel(f'{print_corr_name(correlation)} correlation')
+
+
+# In[10]:
+
+
+mean_corr_gene_df = (corr_cancer_type_df
+    .groupby('gene')
+    .agg(np.mean)
+    .drop(columns=[pval_column])
+)
+
+mean_corr_gene_df.sort_values(by=corr_column, ascending=False).head()
+
+
+# In[11]:
+
+
+mean_corr_gene_df.sort_values(by=corr_column, ascending=False).tail()
+
+
+# In[12]:
+
+
+sns.set(style='ticks', rc={'figure.figsize': (10, 6), 'axes.grid': True, 'axes.grid.axis': 'y'})
+
+with sns.plotting_context('notebook', font_scale=1.2):
+    plt.xlim(-1.0, 1.0)
+    sns.histplot(data=mean_corr_gene_df, x=corr_column)
+    plt.axvline(0.0, linestyle=':', color='black')
+    plt.axvline(mean_corr_gene_df[corr_column].mean(), linestyle='--', color='blue')
+    plt.title(
+        f'Distribution of average {print_corr_name(correlation)} correlations between model size and performance, per gene',
+        y=1.02
+    )
+    plt.xlabel(f'Mean {print_corr_name(correlation)} correlation')
+    plt.ylabel('Gene count')
+
+print(f'Mean of mean correlations: {mean_corr_gene_df[corr_column].mean():.4f}')
+
+
+# In[13]:
+
+
+mean_corr_cancer_type_df = (corr_cancer_type_df
+    .groupby('cancer_type')
+    .agg(np.mean)
+    .drop(columns=[pval_column])
+)
+
+mean_corr_cancer_type_df.sort_values(by=corr_column, ascending=False).head()
+
+
+# In[14]:
+
+
+mean_corr_cancer_type_df.sort_values(by=corr_column, ascending=False).tail()
+
+
+# In[15]:
+
+
+sns.set(style='ticks', rc={'figure.figsize': (10, 6), 'axes.grid': True, 'axes.grid.axis': 'y'})
+
+with sns.plotting_context('notebook', font_scale=1.2):
+    plt.xlim(-1.0, 1.0)
+    sns.histplot(data=mean_corr_cancer_type_df, x=corr_column)
+    plt.axvline(0.0, linestyle=':', color='black')
+    plt.axvline(mean_corr_cancer_type_df[corr_column].mean(), linestyle='--', color='blue')
+    plt.title(f'Distribution of average {print_corr_name(correlation)} correlations between model size and performance, per cancer type', y=1.02)
+    plt.xlabel(f'Mean {print_corr_name(correlation)} correlation')
+    plt.ylabel('Cancer type count')
+
+print(f'Mean of mean correlations: {mean_corr_cancer_type_df[corr_column].mean():.4f}')
```

Comment:
This graph really confused me at first. You're looking at the average correlation across all different genes that could be predicted for each cancer type? What do you make of these results? Also, couldn't that be biased since there are some gene/cancer type pairs you're skipping? It would really help to add a little contextualization of this graph below.

Reply:
Good question! This is definitely confusing, I'll add some explanation.

Like you said, in this plot I'm first taking the mean correlation per gene/cancer type combination, then averaging over genes for each cancer type. And yes, this could be biased if some cancer types are frequently skipped or skipped in certain genes, but on average I think it gives a coarse-grained idea of which cancer types really benefit from larger models, and which benefit more from regularized/smaller models.

I'll also add the gene count for each cancer type, just to see if there are any that seem to be very over/under-represented.